File: python/difference-of-squares/difference_of_squares.py
def square_of_sum(n):
    return sum(x for x in range(1, n+1)) ** 2


def sum_of_squares(n):
    return sum(x**2 for x in range(1, n+1))


def difference_of_squares(n):
    return square_of_sum(n) - sum_of_squares(n)


if __name__ == "__main__":
    print(f"{sum_of_squares(1)=}")
    print(f"{sum_of_squares(2)=}")
    print(f"{square_of_sum(1)=}")
    print(f"{square_of_sum(2)=}")

File: go/difference-of-squares/difference_of_squares.go
// Package diffsquares implements a solution for the exercism difference-of-squares challenge
package diffsquares

// SquareOfSum returns the square of the sum of the first n natural numbers
func SquareOfSum(n int) int {
	sum := n * (n + 1) / 2
	return sum * sum
}

// SumOfSquares returns the sum of the squares of the first n natural numbers
func SumOfSquares(n int) int {
	return (2*n + 1) * n * (n + 1) / 6
}

// Difference returns the difference between SquareOfSum and SumOfSquares
func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}
